components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -418,6 +418,7 @@ public static boolean jsFunction_manageAPI(Context cx, Scriptable thisObj, Objec
             JSONParser parser = new JSONParser();
             properties = (JSONObject) parser.parse(additionalProperties);
         }
+        String authorizationHeader = (String) apiData.get("authorizationHeader", apiData);
```

Comment:
Shall we move authorizationHeader to constant as well?

Reply:
This is the place where the api details are obtained from the html form to be set to the API object. None of these keys ("apiName", "provider", "authorizationHeader") are added as constants.

Reply:
We normally do not keep strings like this in the middle of the code block. Instead we move them to constants.